### DISCUSSION TITLE
How to clean dependabot PRs when body is used as merge message?

### DISCUSSION THREAD

**Discussion by tomdavidson** [chdsbd/kodiak#843] [Q&A] [answered]
Hello, I have kodiak and dependabot. Its fairly slick - thank you. On catch is that dependabot has PR bodies with lots of info and markup that makes my git log kinda messy. Here's a [gist as an example ](https://gist.github.com/tomdavidson/0730e60d709463f2b5c9c9ae17a1aeeb) of what shows in the commit log.

My [full kodiak.toml is in this gist](https://gist.github.com/tomdavidson/2bdb9d7eef5456b9e77a1cfb81e5cefc), but  I am stripping the html form the comments:
```toml
[merge.message]
# works with squash commits and semeantic PR.
title = "pull_request_title"
body = "pull_request_body"
strip_html_comments = true
```

The informative dependabot PR body message is helpful ... just not right for the merge message. Can we have different messages per user? Any suggestions?

**Comment by sbdchd** [comment, marked as answer]
Sorry for the delay! Do you think the `cut_body_after` and `cut_body_before` params could be of use here?

I think there are some html comments in dependabot's pr descriptions that can be used for the markers

https://kodiakhq.com/docs/config-reference#mergemessagecut_body_before

**Comment by tomdavidson** [comment]
@sbdchd thanks. Yes, That is one path, but I am using those with our own PR templates. I think I could add some string from dependabot PR and add it to the PR so I can use the `cut_body_*` with both. Maybe different merge settings for different users is worth looking into?

**Reply by sbdchd** [reply]
Oh so you are already using the `cut_body_*` stuff to remove other content?

hmm, I think we could have it be per user, maybe we could shoehorn a way to have multiple sets, like `cut_body_before` could be `list[str] | str` -- although not sure how we'd match up a given pair -- maybe it would be better as a regex?

**Reply by sbdchd** [reply]
related discussion from when `cut_body_*` was originally added: https://github.com/chdsbd/kodiak/issues/550

**Reply by tomdavidson** [reply]
Ha nice call back to dependabot in #550!  I think the `cut_body_` lists could actually be more relevant for the itches I'm scratchin than different merge settings based on username - and I'm assuming it would be simpler and not necessarily mutually exclusive with future per user setting.

How about kodiak function treats the setting as a list, even if its a list of one, converting the str to a list for backward compatibility. No need to match up or have equal size lists for `_after` and `_before`, treat each element as logical OR and greedy - the largest cut wins ?

```toml
cut_body_before = ["a", "b", "c"]
cut_body_after = "wyoming"
```

```python
# I dont know py, but something like this looks like it could shoehorn well
cut_index = functools.reduce(lambda a, b: a if a > b else b, cut_body_before_list))

```

**Reply by sbdchd** [reply]
I think people might be surprised by largest cut wins, maybe it would be better to have another setting:

```python
cut_body: list[tuple[str, str]]
```

so then multiple start/end pairs could be passed:

```toml
cut_body = [["start_a", "end_a"], ["start_b", "end_b"]]
```

**Reply by tomdavidson** [reply]
It would be hard to be surprised by the tuples.

**Reply by chdsbd** [reply]
Should we open a new issue to track a feature request for this @tomdavidson?
